flecsi/log/state.hh

```diff
@@ -144,13 +149,20 @@ public:
               << FLOG_COLOR_PLAIN << std::endl;
 #endif
 
+#if FLECSI_RUNTIME_MODEL != FLECSI_RUNTIME_MODEL_hpx
     {
       int p, np;
       MPI_Comm_rank(MPI_COMM_WORLD, &p);
       MPI_Comm_size(MPI_COMM_WORLD, &np);
       process_ = p;
       processes_ = np;
     }
+#else
+    {
+      process_ = ::hpx::get_locality_id();
+      processes_ = ::hpx::get_num_localities(::hpx::launch::sync);
```

Comment:
Can we avoid this bifurcation and redundancy (with `*/context.cc`) by adding `util` functions for these trivial queries?

Reply:
I think for now, we can avoid using HPX functions here altogether as the logging library relies on MPI anyways. In the future we might want to use HPX logging facilities instead, but this might be better done in a separate PR.

Reply:
Well, that's kicking the can down the road, but probably in the right direction.  We could do the same thing with the whole logging facility, and that might be more useful.